metric_tank/mdata/aggmetric.go

```diff
@@ -150,11 +150,8 @@ func (a *AggMetric) getChunkByT0(ts uint32) *chunk.Chunk {
 }
 
 func (a *AggMetric) getChunk(pos int) *chunk.Chunk {
-	if pos < 0 {
-		return nil
-	}
-	if pos >= len(a.Chunks) {
-		return nil
+	if pos < 0 || pos >= len(a.Chunks) {
+		panic(fmt.Sprintf("aggmetric %s queried for chunk %d out of %d chunks", a.Key, pos, len(a.Chunks)))
```

Comment:
i understand you point of view on panics, but the consequence (10+ customer impacting outages in 2days) is far too great.   If we were deploying fixes after the first panic that would be fine.  But as it has already been 48hours and without a fix deployed it is clear that approach of allowing the code to panic and then fixing it when it does is not working.